modules/lambdas/main.tf

```diff
@@ -299,25 +299,44 @@ resource "aws_lambda_function" "post_copy_request_to_queue" {
     subnet_ids         = var.lambda_subnet_ids
     security_group_ids = [module.lambda_security_group.vpc_postgres_ingress_all_egress_id]
   }
+  environment {
+    variables = {
+      PREFIX             = var.prefix
+      DATABASE_PORT      = var.database_port
+      DATABASE_NAME      = var.database_name
+      DATABASE_USER      = var.database_app_user
+      DB_QUEUE_URL       = var.orca_sqs_status_update_queue_id
+      RECOVERY_QUEUE_URL = var.orca_sqs_recovery_queue_id
+    }
+  }
 }
 
 # Permissions to allow S3 trigger to invoke lambda
```

Comment:
We should be able to construct the arns so no need to have them in a variable. see below. Also may want to change allow_bucket to something more informative like "allow_s3_trigger"

```
resource "aws_lambda_permission" "allow_s3_trigger" {
  for_each = toset(local.orca_buckets)
  statement_id  = "AllowExecutionFromS3"
  action        = "lambda:InvokeFunction"
  function_name = aws_lambda_function.post_copy_request_to_queue.function_name
  principal     = "s3.amazonaws.com"
  source_arn    = "arn:aws:s3:::${each.value}"
}
```

Note that the arn list could also be created in the local block and referenced here as another option.

Reply:
interesting! I didn't know we can use that way. Good thing to learn